### PR TITLE
Zcv2 707, 706 Impl

### PR DESCRIPTION
**Pull request by divyag1811**
fetching the recent trips from redis
populating data into booking collection

### CODE REVIEW

File: src/main/java/com/zed/booking/config/BookingScheduler.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+@Component
+public class BookingScheduler {
+
+	@Autowired
+	MongoTemplate mongoTemplate;
+	
+	@Autowired
+	BookingService bookingService;
+
+	//@Scheduled(cron = "0 */2 * * * ?")
```
Comment (meetanshug) [inline src/main/java/com/zed/booking/config/BookingScheduler.java:27]: PLease put a comment that this is for reconciliation

File: src/main/java/com/zed/booking/entity/Booking.java [hunk 2]
```diff
@@ -42,4 +51,5 @@ public class Booking {
     private DriverDetails driverDetails;
     private FareDetails fareDetails;
     private JourneyDetails journeyDetails;
+    private Boolean isInvoiceSent;
```
Comment (meetanshug) [inline src/main/java/com/zed/booking/entity/Booking.java:54]: Move this out of booking

File: src/main/java/com/zed/booking/service/BookingService.java [hunk 3]
```diff
@@ -2,71 +2,140 @@
...
+			cacheService.updateBookingForUser(journey.getRefCustomerId(), bookingObj);
+		} else {
+			throw new BadRequestException(MessageConstants.NOT_TERMINAL_STATUS);
+		}
+		
+	}
 
-    @Autowired
-    BookingUtil bookingUtil;
+	public Page<Journey> fetchAllBookings(String customerId, int page, int size, Sort.Direction sortDirection) {
```
Comment (meetanshug) [inline src/main/java/com/zed/booking/service/BookingService.java:76]: Please check this

File: src/main/java/com/zed/booking/service/CacheService.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	private static final int RECENT_TRIPS_CACHE_LIMIT = 5;
+
+	@Autowired
+	BookingRepository bookingRepository;
+
+	@Cacheable(value = "pastTrips", key = "#customerID")
+	public List<Booking> getPastTrips(String customerID) {
+		log.info("Fetching past trips for user {}", customerID);
+		// Fetch trips from database if not found in cache
+		List<Booking> fetchTripsFromDatabase = fetchTripsFromDatabase(customerID);
```
Comment (meetanshug) [inline src/main/java/com/zed/booking/service/CacheService.java:44]: fetchTripsFromCache()

File: src/main/java/com/zed/booking/util/BookingUtil.java [hunk 5]
```diff
@@ -44,24 +46,24 @@ public void reqValidation(Booking bookingReq) {
 
     private static void validateJourneyDetails(JourneyDetails journeyDetails) {
```
Comment (meetanshug) [inline src/main/java/com/zed/booking/util/BookingUtil.java:47]: recheck if this is needed

File: src/main/java/com/zed/booking/util/Status.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.zed.booking.util;
```
Comment (meetanshug) [inline src/main/java/com/zed/booking/util/Status.java:1]: needed?

File: src/main/java/com/zed/booking/service/CacheService.java (hunk 4 above)
Comment (divyag1811) [reply src/main/java/com/zed/booking/service/CacheService.java:44]: done

File: src/main/java/com/zed/booking/entity/Booking.java (hunk 2 above)
Comment (divyag1811) [reply src/main/java/com/zed/booking/entity/Booking.java]: done

File: src/main/java/com/zed/booking/util/BookingUtil.java (hunk 5 above)
Comment (divyag1811) [reply src/main/java/com/zed/booking/util/BookingUtil.java]: removed unused methods

File: src/main/java/com/zed/booking/util/Status.java (hunk 6 above)
Comment (divyag1811) [reply src/main/java/com/zed/booking/util/Status.java:1]: removed

**Review by meetanshug — APPROVED** [review]

**Pull request merged**